pytest_rts/utils/common.py

```diff
@@ -32,6 +41,34 @@ def get_existing_tests(coverage_file_path: str) -> Set[str]:
     }
 
 
+def get_tests_from_changes(
+    file_diffs: Dict[str, str], coverage_file_path: str
+) -> Set[str]:
+    """Returns the test set from given Git diffs"""
+    coverage_data = CoverageData(coverage_file_path)
+    coverage_data.read()
+
+    tests: Set[str] = set()
+    for changed_file in file_diffs:
+        changed_lines = get_changed_lines(file_diffs[changed_file])
+        for line_number in coverage_data.contexts_by_lineno(changed_file):
+            if line_number in changed_lines:
+                tests.update(
+                    strip_pytest_cov_testname(testname)
+                    for testname in coverage_data.contexts_by_lineno(changed_file)[
+                        line_number
+                    ]
+                )
+    return tests
+
+
+def get_tests_current(coverage_file_path: str) -> Set[str]:
+    """Returns the test set from working directory changes"""
+    changed_files = get_changed_files_current()
+    file_diffs = get_file_diff_dict_current(changed_files)
+    return get_tests_from_changes(file_diffs, coverage_file_path)
```

Comment:
I now passed the `GitRepository` object around in the functions to prevent the extra call. Getting all the changed lines in one diff call could be possible but requires a lot of parsing, I think. For example changing 2 files:
```
(.venv) eero@eero-thinkpad:~/Desktop/pytest-rts$ git diff -U0
diff --git a/pytest_rts/utils/common.py b/pytest_rts/utils/common.py
index 8174a3a..4d8e7ba 100644
--- a/pytest_rts/utils/common.py
+++ b/pytest_rts/utils/common.py
@@ -60,0 +61 @@ def get_tests_from_changes(
+
@@ -64,0 +66 @@ def get_tests_from_changes(
+
@@ -82,0 +85 @@ def strip_pytest_cov_testname(testname: str) -> str:
+        
diff --git a/pytest_rts/utils/git.py b/pytest_rts/utils/git.py
index 6283ba4..2c573ad 100644
--- a/pytest_rts/utils/git.py
+++ b/pytest_rts/utils/git.py
@@ -10,0 +11 @@ def get_changed_files_current(repo: GitRepository) -> List[str]:
+
@@ -28,0 +30 @@ def get_changed_lines(diff: str) -> Set[int]:
+
@@ -49,0 +52 @@ def get_changed_lines(diff: str) -> Set[int]:
+            
```

Are git calls that slow that it would make sense to start parsing the testfile names from this sort of output?

Reply:
I have no evidence that it is slow, but let's keep an eye on it. No need to refactor it now.